backend.go

```diff
@@ -128,6 +133,54 @@ func (r *RedisBackend) IncrementID(ctx context.Context, id int) (int64, error) {
 	return cnt, nil
 }
 
+func (r *RedisBackend) GetMessageReply(ctx context.Context, msg MessageIdentifier) ([]Message, error) {
+	log.Info().Str("return_queue", msg.Retqueue).Msg("Waiting reply")
+	responses := []Message{}
+	retQueueLen, err := r.client.LLen(ctx, msg.Retqueue).Result()
+	if err != nil {
+		log.Error().Err(err).Msg("error fetching from redis")
+		return responses, err
+	}
+	// TODO: check if the message ready
+	if int(retQueueLen) < 1 {
+		return nil, errors.New("The message not handled yet, it is in progress.")
+	} else {
```

Comment:
the branching here can be avoided. Instead of asking for the length and checking against it, you can:
```go
results, err := r.client.LRange(ctx, msg.Retqueue, 0, -1).Result()
if len(results) == 0 {
   // nothing found
}
```